src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,144 @@
+        Node<K, V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            return;
+        }
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                size--;
+                return;
+            }
+            if (node.next == null) {
+                break;
+            }
+            node = node.next;
+        }
+        node.next = new Node<>(key, value, null);
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
+    @Override
+    public List<V> getValues() {
+        Node<K, V>[] nodes = getAllNodes();
+        List<V> values = new ArrayList<>();
+        for (Node<K, V> node : nodes) {
+            values.add(node.value);
+        }
+        return values;
+    }
+
+    @Override
+    public Set<K> getKeys() {
+        Node<K, V>[] nodes = getAllNodes();
+        Set<K> keys = new HashSet<>();
+        for (Node<K, V> node : nodes) {
+            keys.add(node.key);
+        }
+        return keys;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V>[] getAllNodes() {
+        Node<K, V>[] nodes = new Node[size];
+        int index = 0;
+        for (Node<K, V> node : table) {
+            if (node != null) {
+                nodes[index] = node;
+                index++;
+                while (nodes[index - 1].next != null) {
+                    nodes[index] = nodes[index - 1].next;
+                    index++;
+                }
+            }
+        }
+        return nodes;
+    }
+
+    private void checkThreshold() {
+        if (size > (int) (LOAD_FACTOR * table.length)) {
+            resize();
+        }
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        final Node<K, V>[] nodes = getAllNodes();
+        table = new Node[table.length * 2];
+        transfer(nodes);
+    }
+
+    private void transfer(Node<K,V>[] nodes) {
+        for (Node<K, V> currentNode : nodes) {
+            int index = getIndex(currentNode.key);
+            if (table[index] == null) {
+                table[index] = new Node<>(currentNode.key, currentNode.value, null);
+                continue;
+            }
+            Node<K, V> node = table[index];
+            while (node != null) {
+                if (node.next == null) {
+                    node.next = new Node<>(currentNode.key, currentNode.value, null);
+                    break;
+                }
+                node = node.next;
+            }
```

Comment:
you can call inside of while loop put method for each node, after this you won't need 117-121 lines